Error while checking build/O2/o2-dataflow for 4d95e1431160d39b244ecd929d695a561b6923c4 at 2022-09-02 16:53:
```
## sw/BUILD/O2-latest/log
100% tests passed, 0 tests failed out of 423
 88/106 Test #102: test_Framework_test_SimpleDataProcessingDevice01 ........***Failed    2.85 sec
[ERROR] pid 24559 (internal-dpl-injected-dummy-sink) crashed with 1
[ERROR] SEVERE: Device internal-dpl-injected-dummy-sink (24559) returned with 1
99% tests passed, 1 tests failed out of 101
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/4d95e1431160d39b244ecd929d695a561b6923c4/build_O2_o2-dataflow/pretty.html).

--- a/DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h
+++ b/DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h
@@ -112,6 +112,7 @@ class GRPECSObject
   // methods to manipulate the list of FLPs in the run
   const std::vector<unsigned short>& getListOfFLPs() const { return mFLPs; }
   void addFLP(unsigned short flp) { mFLPs.push_back(flp); }
+  void setListOfFLPs(std::vector<unsigned short> listFLPs) { mFLPs = listFLPs; }
   bool getFLPStatus(unsigned short flp) const
   {
     return std::find(mFLPs.begin(), mFLPs.end(), flp) == mFLPs.end() ? false : true;

--- a/DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h
+++ b/DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h
@@ -112,7 +112,7 @@ class GRPECSObject
   // methods to manipulate the list of FLPs in the run
   const std::vector<unsigned short>& getListOfFLPs() const { return mFLPs; }
   void addFLP(unsigned short flp) { mFLPs.push_back(flp); }
-  void setListOfFLPs(std::vector<unsigned short> listFLPs) { mFLPs = listFLPs; }
+  void setListOfFLPs(const std::vector<unsigned short>& listFLPs) { mFLPs = listFLPs; }
   bool getFLPStatus(unsigned short flp) const
   {
     return std::find(mFLPs.begin(), mFLPs.end(), flp) == mFLPs.end() ? false : true;

--- a/Detectors/GRP/workflows/src/create-grp-ecs.cxx
+++ b/Detectors/GRP/workflows/src/create-grp-ecs.cxx
@@ -53,12 +53,14 @@ void createGRPECSObject(const std::string& dataPeriod,
   }
   auto detMaskCont = detMask & o2::detectors::DetID::getMask(detsContinuousRO);
   auto detMaskTrig = detMask & o2::detectors::DetID::getMask(detsTrigger);
-  std::stringstream flpListSS(flpList);
-  std::vector<unsigned short> flps;
-  while (flpListSS.good()) {
-    string substr;
-    getline(flpListSS, substr, ',');
-    flps.push_back(atoi(substr.c_str()));
+  auto flpsVec = o2::utils::Str::tokenize(flpList, ',');
+  for (const auto& s : flpsVec) {
+    try {
+      addFlp((unsigned short)std::stoi(s));
+    }
+    catch(const std::exception& e) {
+      LOG(alarm) << "could not convert string " << s << " to integer FLP ID, error : " << e.what(); 
+    }
   }
   LOG(info) << tstart << " " << tend;
   if (tstart == 0) {
@@ -81,7 +83,6 @@ void createGRPECSObject(const std::string& dataPeriod,
   grpecs.setRun(run);
   grpecs.setRunType((GRPECSObject::RunType)runType);
   grpecs.setDataPeriod(dataPeriod);
-  grpecs.setListOfFLPs(flps);
 
   grpecs.print();
   std::map<std::string, std::string> metadata;

--- a/Detectors/GRP/workflows/src/create-grp-ecs.cxx
+++ b/Detectors/GRP/workflows/src/create-grp-ecs.cxx
@@ -35,6 +35,7 @@ void createGRPECSObject(const std::string& dataPeriod,
                         const std::string& detsReadout,
                         const std::string& detsContinuousRO,
                         const std::string& detsTrigger,
+                        const std::string& flpList,
                         long tstart,
                         long tend,
                         long marginAtSOR,
@@ -52,6 +53,13 @@ void createGRPECSObject(const std::string& dataPeriod,
   }
   auto detMaskCont = detMask & o2::detectors::DetID::getMask(detsContinuousRO);
   auto detMaskTrig = detMask & o2::detectors::DetID::getMask(detsTrigger);
+  std::stringstream flpListSS(flpList);
+  std::vector<unsigned short> flps;
+  while (flpListSS.good()) {
+    string substr;
+    getline(flpListSS, substr, ',');
+    flps.push_back(atoi(substr.c_str()));
+  }
   LOG(info) << tstart << " " << tend;
   if (tstart == 0) {
     tstart = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
@@ -73,6 +81,7 @@ void createGRPECSObject(const std::string& dataPeriod,
   grpecs.setRun(run);
   grpecs.setRunType((GRPECSObject::RunType)runType);
   grpecs.setDataPeriod(dataPeriod);
+  grpecs.setListOfFLPs(flps);
 
   grpecs.print();
   std::map<std::string, std::string> metadata;
@@ -163,6 +172,7 @@ int main(int argc, char** argv)
     add_option("detectors,d", bpo::value<string>()->default_value("all"), "comma separated list of detectors");
     add_option("continuous,c", bpo::value<string>()->default_value("ITS,TPC,TOF,MFT,MCH,MID,ZDC,FT0,FV0,FDD,CTP"), "comma separated list of detectors in continuous readout mode");
     add_option("triggering,g", bpo::value<string>()->default_value("FT0,FV0"), "comma separated list of detectors providing a trigger");
+    add_option("flps,f", bpo::value<string>()->default_value(""), "comma separated list of FLPs in the data taking");
     add_option("start-time,s", bpo::value<long>()->default_value(0), "run start time in ms, now() if 0");
     add_option("end-time,e", bpo::value<long>()->default_value(0), "run end time in ms, start-time+3days is used if 0");
     add_option("ccdb-server", bpo::value<std::string>()->default_value("http://alice-ccdb.cern.ch"), "CCDB server for upload, local file if empty");
@@ -220,6 +230,7 @@ int main(int argc, char** argv)
     vm["detectors"].as<std::string>(),
     vm["continuous"].as<std::string>(),
     vm["triggering"].as<std::string>(),
+    vm["flps"].as<std::string>(),
     vm["start-time"].as<long>(),
     vm["end-time"].as<long>(),
     vm["marginSOR"].as<long>(),

